src/components/HexagonGrid.tsx

```diff
@@ -82,10 +87,30 @@ function HexagonGrid(props: HexagonGridProps) {
   };
 
   function testOnClick(hexagon: HexData) {
```

Comment:
since this is no longer the test function, and we're actually doing this onClick, let's name it better.

Also, don't name functions based on how they're called. Name them for what they do. Maybe it's "updateHexagon"